kernel/src/main/scala-3.x/cats/kernel/derivation/EqDerivation.scala

```diff
@@ -0,0 +1,43 @@
+
+package cats.kernel.derivation
+
+import cats.kernel.Eq
+import scala.deriving._
+
+import scala.compiletime.{erasedValue, summonInline, constValue}
+
+inline private[derivation] def summonAll[T <: Tuple, F[_]]: List[F[Any]] =
```

Comment:
Isn't that `scala.compileTime.summonAll[Tuple.Map[T, F]].toList`?

I believe you could also drop the `toList`, tuples are pretty powerful in Scala 3, even zipping is supported.

Anyway, I very much like the initiative of adding this to cats and will watch this PR (and stop nitpicking now 😉)!

Reply:
You're probably right :) I've copied my older code verbatim to test if this is binary compatible and if it works, so now I'm leisurely researching all the ways to make this more concise and faster :)

Reply:
I managed to remove the lists and iterators and instead only use the tuples (plus inlining the whole thing).

Now there's a concern about inlining the `derived` call when deriving the coproduct, not sure if this can have any repercussions